src/code/cutscene_camera.c

```diff
@@ -353,28 +357,576 @@ void CutsceneCamera_Reset(void) {
+        interpState->waypoint = 0;
+        interpState->curFrame = 0;
+        interpState->duration = 1;
+        if (camFov != NULL) {
+            *camFov = miscCmd->fov;
+        }
+        if (camRoll != NULL) {
+            *camRoll = CAM_DEG_TO_BINANG(miscCmd->roll);
+        }
+    }
+
+    if (camPos != NULL) {
+        camPos->x = pointCmd->pos.x;
+        camPos->y = pointCmd->pos.y;
+        camPos->z = pointCmd->pos.z;
+    }
+
+    interpState->curFrame++;
+
+    if (interpState->curFrame >= pointCmd->duration) {
+        // Finished
+        interpState->type = CS_CAM_INTERP_OFF;
+        return 1;
+    }
+
+    return 0;
+}
+
+void func_801624EC(f32 u, f32* coeff) {
+    f32 u1 = 1.0f - u;
+
+    coeff[0] = u1 * u1 * 0.5f;
+    coeff[1] = u * u1 + 0.5f;
+    coeff[2] = u * u * 0.5f;
+}
+
+s16 CutsceneCamera_Interp_MultiPointQuadratic(Vec3f* camPos, f32* camFov, s16* camRoll, CsCmdCamPoint* pointCmd,
+                                              CsCmdCamMisc* miscCmd, CutsceneCameraInterp* interpState) {
+    f32 new_var;
+    f32 coeff[3];
+    s32 sp3C[3];
+
+    if (interpState->type != CS_CAM_INTERP_MP_QUAD) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_MP_QUAD;
+        interpState->waypoint = 0;
+        interpState->duration = pointCmd->duration;
+        interpState->curFrame = 0;
+    }
+
+    new_var = (f32)interpState->curFrame / pointCmd[interpState->waypoint + 1].duration;
+
+    if (interpState->waypoint < (interpState->duration - 1)) {
+        sp3C[0] = interpState->waypoint;
+    } else {
+        sp3C[0] = interpState->duration - 1;
+    }
+
+    if ((interpState->waypoint + 1) < (interpState->duration - 1)) {
+        sp3C[1] = interpState->waypoint + 1;
+    } else {
+        sp3C[1] = interpState->duration - 1;
+    }
+
+    if ((interpState->waypoint + 2) < (interpState->duration - 1)) {
+        sp3C[2] = interpState->waypoint + 2;
+    } else {
+        sp3C[2] = interpState->duration - 1;
+    }
+
+    func_801624EC(new_var, coeff);
+
+    if (camPos != NULL) {
+        camPos->x = (coeff[0] * pointCmd[sp3C[0]].pos.x) + (coeff[1] * pointCmd[sp3C[1]].pos.x) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.x);
+        camPos->y = (coeff[0] * pointCmd[sp3C[0]].pos.y) + (coeff[1] * pointCmd[sp3C[1]].pos.y) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.y);
+        camPos->z = (coeff[0] * pointCmd[sp3C[0]].pos.z) + (coeff[1] * pointCmd[sp3C[1]].pos.z) +
+                    (coeff[2] * pointCmd[sp3C[2]].pos.z);
+    }
+
+    if (camFov != NULL) {
+        *camFov =
+            (coeff[0] * miscCmd[sp3C[0]].fov) + (coeff[1] * miscCmd[sp3C[1]].fov) + (coeff[2] * miscCmd[sp3C[2]].fov);
+    }
+
+    if (camRoll != NULL) {
+        s16 sp34[3];
+        s32 sp28[2];
+        s32 temp;
+
+        sp34[0] = CAM_DEG_TO_BINANG(miscCmd[sp3C[0]].roll);
+        sp34[1] = CAM_DEG_TO_BINANG(miscCmd[sp3C[1]].roll);
+        sp34[2] = CAM_DEG_TO_BINANG(miscCmd[sp3C[2]].roll);
```

Comment:
This makes `sp34` seem like a list of "target rolls"? Though I'm not sure what the 0, 1, and 2 are supposed to indicate. Is 0 the "current" point, and 1 and 2 the "next"?

Reply:
`sp34` is an index to the command list below, so I think `waypoint` would actually make more sense here? This is taken from test.6's `sDoubleSoTCsCamData`, but here's the data:
```c
/* 0x0 */ CS_CAM_MISC(15, 0xA, 70, 0), // `miscCmd[0]`
/* 0x1 */ CS_CAM_MISC(15, 0xA, 70, 0), // `miscCmd[1]`
/* 0x2 */ CS_CAM_MISC(15, 0xA, 70, 0), // `miscCmd[2]`
```

Reply:
You mean sp3C is an index? (Not sp34 ?) 🤔

Reply:
Oh whoops, yeah was looking at the wrong variable here. Yeah, 0 is current point and 1/2 are the next two points in the list. 
`sp34` -> `targetRolls`

And yeah, with `targetRolls` being an array here, and `rollDiffToTarget` being the final value in this function, it makes me think those temps aren't fake in the simpler functions, and that it's broken up with temps that align with these more complex/higher order functions